### ISSUE TITLE
Only verify `@MustCallAlias` at normal exits

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#5597]
Consider the following (contrived and rough) example:
```java
@InheritableMustCall("close")
class Test {
  @Owning InputStream is;
  @MustCallAlias Test(@MustCallAlias InputStream p, boolean b) throws SomeException {
    if (b) { throw new SomeException(); }
    this.is = p;
  }
  @EnsuresCalledMethods("this.is", "close")
  public void close() {
    this.is.close();
  }
}
```
Right now, the Resource Leak Checker will report a `mustcallalias.out.of.scope` warning for the `Test` constructor, since when the method throws an exception, `p` is not written into the owning field `this.is`.  This warning is unexpected, as most users would expect the `@MustCallAlias` relationship to hold only when the constructor exits normally, not on both normal and exceptional exits.

We should enhance the Resource Leak Checker so that it only verifies `@MustCallAlias` relationships at normal exits, so no warning would be issued for the above example.  We'll also need to enhance the `MustCallConsistencyAnalyzer` to only assume `@MustCallAlias` relationships on normal (not exceptional) successors of relevant calls.

/cc @kelloggm @mernst @Nargeshdb
